container-engine-lib/lib/backend_impls/docker/docker_kurtosis_backend/backend_creator/backend_creator.go

```diff
@@ -240,13 +240,17 @@ func getDockerKurtosisBackend(
 			return nil, stacktrace.Propagate(err, "An error occurred while getting the logs collector object for enclave '%v'; This is a bug in Kurtosis", enclaveUuid)
 		}
 
-		reverseProxy, err := reverse_proxy_functions.GetReverseProxy(ctx, dockerManager)
+		maybeReverseProxy, err := reverse_proxy_functions.GetReverseProxy(ctx, dockerManager)
 		if err != nil {
 			return nil, stacktrace.Propagate(err, "An error occurred while getting the reverse proxy, This is a bug in Kurtosis")
 		}
-		reverseProxyEnclaveNetworkIpAddress, found := reverseProxy.GetEnclaveNetworksIpAddress()[network.GetId()]
+		if maybeReverseProxy == nil {
+			return nil, stacktrace.NewError("Expected to get the Kurtosis reverse proxy but it was not found, This is a bug in Kurtosis")
```

Comment:
the PR avoid the code panicking but still throws an error if the `GetReverseProxy` is nil. Not sure if this is still back compatible with existing enclaves.

Reply:
I checked retro compatibility with 0.85.48 and 0.85.46, nothing failed while running `kurtosis engine restart` 
0.85.48  -> https://www.loom.com/share/42510b78f4e6486486d9d1a6a5948369
0.85.46  -> https://www.loom.com/share/c4e692bd184c4f4ca9669aee6dcd2b71